anyway/widgets/suburban_widgets/killed_and_injured_count_per_age_group_widget.py

```diff
@@ -0,0 +1,201 @@
+from collections import defaultdict
+from datetime import datetime
+from typing import Callable, Dict, Tuple, List
+
+from flask_babel import _
+from flask_sqlalchemy import BaseQuery
+from sqlalchemy import func, asc
+
+from anyway.app_and_db import db
+from anyway.backend_constants import BE_CONST, InjurySeverity as IS
+from anyway.models import InvolvedMarkerView
+from anyway.request_params import RequestParams
+from anyway.utilities import parse_age_from_range
+from anyway.widgets.suburban_widgets.sub_urban_widget import SubUrbanWidget
+from anyway.widgets.widget import register
+from anyway.widgets.widget_utils import (
+    order_by_list_stack_bar_widget,
+    gen_entity_labels,
+    add_empty_keys_to_gen_two_level_dict,
+)
+
+# RequestParams is not hashable, so we can't use functools.lru_cache
+cache_dict = {}
```

Comment:
I think the cache will only be relevant in development environment.

Reply:
Why? the two widgets use the same data from the DB so instead of fetching and processing it twice i am caching it

Reply:
I do not mean to drop the cooperation between the two classes, only to achieve it in a way that will allow us to keep the standard of single class in a file. There are several options:
1 - have both classes inherit from a base class that provides the local cache functionality.
2 - have the cache functionality in a utility class that both use as a member.
3 - have the cache functionality implemented as a function in widget utils

Reply:
As we mention the cache, the `popitem()` works in a LIFO order, and I think here we need FIFO.